common/util/FileUtil.cpp

```diff
@@ -0,0 +1,36 @@
+#include "FileUtil.h"
+#include <iostream>
+#include <stdio.h>  /* defines FILENAME_MAX */
+
+#ifdef _WIN32
+  #include <direct.h>
+  #define GetCurrentDir _getcwd
+#else
+  #include <unistd.h>
+  #define GetCurrentDir getcwd
+#endif
+
+std::string FileUtil::GetCurrentWorkingDir() {
+  char buff[FILENAME_MAX];
+  GetCurrentDir(buff, FILENAME_MAX);
+  std::string current_working_dir(buff);
+  return current_working_dir;
+}
+
+std::string FileUtil::get_file_path(std::vector<std::string> input) {

```

Comment:
This is a super minor detail, but if you replace `std::vector<std::string> input` with `const std::vector<std::string>& input` it will make this code slightly faster.  The combination of `const` and `&` tell the C++ compiler that this function isn't going to modify `input`, so instead of making a separate copy of `input` to be used in this function, it will just take a reference to `input`.  In this case it's not a big enough difference to matter (the vector of strings is tiny and fast to copy), but it's a good pattern to get used to using.

Reply:
Thank you for this!